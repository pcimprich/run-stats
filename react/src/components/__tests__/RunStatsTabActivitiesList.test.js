var React = require('react');
var RunStatsTabActivitiesList = require('../RunStatsTabActivitiesList').Component;
var renderer = require('react-test-renderer');

test('RunStatsTabActivitiesList renders a list of activities', () => {
  const runs = {
      	isFetching: false,
      	didInvalidate: false,
	  	lastUpdated: 1439478405547,
		filter: { key: 'yr.2017', count: 145 },
      	items: [
        	{"id":413,"date":"2017/02/21","year":2017,"month":2,"week":8,"time":"11:44:00","distance":10.47,"duration":"01:06:09","pace":"00:06:19","kcal":1000,"elevation":151,"cadence":79,"steps":10420,"location":"Radošovice","source":"Garmin","notes":null},
			{"id":415,"date":"2017/02/20","year":2017,"month":2,"week":8,"time":"24:25:00","distance":6.79,"duration":"00:40:58","pace":"00:06:02","kcal":664,"elevation":61,"cadence":78,"steps":6380,"location":"Radošovice","source":"Garmin","notes":null}
      	],
		groupedByWeek: {
			"2017w15": { count: 5, distance: 35, duration: '4:50:49', kcal: 7023, elevation: 957, maxDistance: 12.3 },
		},
		groupedByMonth: {
			"2017m1": { count: 14, distance: 95, duration: '16:50:49', kcal: 22023, elevation: 5357, maxDistance: 15.2 },
		},
		groupedByYear: {
			"2016": { count: 145, distance: 1067, duration: "108:20:10", elevation: 15886, kcal: 96988, maxDistance: 21.1 },
		},
		groupedTotal: { 
			total: { count: 445, distance: 3067, duration: "408:20:10", elevation: 35886, kcal: 426988, maxDistance: 21.1 }
		},
		isImporting: false,
		lastImported: 1239478406530,
		newActivities: 0,
		showImportModal: false
	};
	
  const component = renderer.create(
	<RunStatsTabActivitiesList runs={runs} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

