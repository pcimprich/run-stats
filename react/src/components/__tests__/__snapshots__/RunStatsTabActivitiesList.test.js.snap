// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RunStatsTabActivitiesList renders a list of activities 1`] = `
<table
  className="table table-striped"
>
  <thead>
    <tr>
      <th />
      <th
        className="right"
      >
        Date
      </th>
      <th
        className="right"
      >
        Distance
      </th>
      <th
        className="right"
      >
        Duration
      </th>
      <th
        className="right"
      >
        Avg Pace
      </th>
      <th
        className="right"
      >
        Elevation Gain
      </th>
      <th
        className="right"
      >
        Energy
      </th>
      <th />
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        <i
          className="material-icons"
        >
          directions_run
        </i>
         
        Radošovice
      </td>
      <td
        className="right"
      >
        2017/02/21
      </td>
      <td
        className="right"
      >
        10.47
         km
      </td>
      <td
        className="right"
      >
        1:06:09
      </td>
      <td
        className="right"
      >
        06:19
      </td>
      <td
        className="right"
      >
        151
         m
      </td>
      <td
        className="right"
      >
        1000
         kcal
      </td>
      <td
        className="right"
      >
        <i
          className="material-icons"
        >
          watch
        </i>
      </td>
    </tr>
    <tr>
      <td>
        <i
          className="material-icons"
        >
          directions_run
        </i>
         
        Radošovice
      </td>
      <td
        className="right"
      >
        2017/02/20
      </td>
      <td
        className="right"
      >
        6.79
         km
      </td>
      <td
        className="right"
      >
        40:58
      </td>
      <td
        className="right"
      >
        06:02
      </td>
      <td
        className="right"
      >
        61
         m
      </td>
      <td
        className="right"
      >
        664
         kcal
      </td>
      <td
        className="right"
      >
        <i
          className="material-icons"
        >
          watch
        </i>
      </td>
    </tr>
  </tbody>
</table>
`;
